politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -404,14 +498,134 @@ func (p *piPlugin) commentWritesAllowed(token []byte) error {
 		ticketvote.VoteStatusStarted:
 		// Comment writes are allowed on these vote statuses
 		return nil
+	case ticketvote.VoteStatusApproved:
+		// Get billing status to determine whether to allow author updates
+		// or not.
+		var bsc pi.BillingStatusChange
+		billingStatuses, err := p.billingStatuses(token)
+		if err != nil {
+			return err
+		}
+		// We assume here that admins can set a billing status only once
+		if len(billingStatuses) > 0 {
+			bsc = billingStatuses[0]
+			if bsc.Status == pi.BillingStatusClosed ||
+				bsc.Status == pi.BillingStatusCompleted {
+				// If billing status is set to closed or completed comment writes
+				// are not allowed.
+				return backend.PluginError{
+					PluginID:  pi.PluginID,
+					ErrorCode: uint32(pi.ErrorCodeBillingStatusInvalid),
+					ErrorContext: "billing status is set to closed/completed;" +
+						" proposal is locked",
+				}
+			}
+		}
+		// Get latest proposal author update
+		gar, err := p.comments(token)
+		if err != nil {
+			return err
+		}
+		latestAuthorUpdate, err := p.latestAuthorUpdate(token, gar.Comments)
+		if err != nil {
+			return err
+		}
+		switch cmd {
+		// If that's a new comment then it must be either a new author
+		// update or a comment on the latest author update thread.
+		case comments.CmdNew:
+			// Decode payload
```

Comment:
I would abstract the code under these switch cases into functions as well to improve readability.